modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/utils/OclLoader.java

```diff
@@ -8,19 +8,92 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.models.billing.RatingMode;
+import org.eclipse.xpanse.modules.models.billing.enums.BillingMode;
 import org.eclipse.xpanse.modules.models.servicetemplate.Ocl;
+import org.eclipse.xpanse.modules.models.servicetemplate.ServiceFlavor;
+import org.eclipse.xpanse.modules.models.servicetemplate.exceptions.InvalidServiceFlavorsException;
 import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
 
 /**
  * Bean to deserialize Ocl data.
  */
+@Slf4j
 @Component
 public class OclLoader {
 
     private final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
 
+    /**
+     * Get Ocl Object from URL.
+     *
+     * @param url URL of Ocl file.
+     * @return Ocl object.
+     * @throws Exception if the Ocl file is invalid.
+     */
     public Ocl getOcl(URL url) throws Exception {
-        return mapper.readValue(url, Ocl.class);
+        Ocl ocl = mapper.readValue(url, Ocl.class);
+        validateServiceFlavors(ocl);
+        return ocl;
+    }
+
+
+    private void validateServiceFlavors(Ocl ocl) {
+        List<String> errors = new ArrayList<>();
+        List<ServiceFlavor> serviceFlavors = ocl.getFlavors().getServiceFlavors();
+        Map<String, Long> nameCountMap = serviceFlavors.stream()
+                .collect(Collectors.groupingBy(ServiceFlavor::getName, Collectors.counting()));
+        nameCountMap.entrySet().stream().filter(entry -> entry.getValue() > 1).forEach(entry -> {
+            String message =
+                    String.format("Service flavor with name %s is duplicated.", entry.getKey());
+            log.error(message);
+            errors.add(message);
+        });
+        List<BillingMode> billingModes = ocl.getBilling().getBillingModes();
```

Comment:
I think this is not a good place for this logic. 
We can add a new component for `BillingConfigValidator` in `servicetemplate` module and use it.